tests/jax/integration/executor_test.py

```diff
@@ -13,265 +13,92 @@
-    executor_observe=executing.FeedforwardExecutorObserve,
-    executor_select_action=executing.FeedforwardExecutorSelectAction,
-    executor_adder=building.ParallelSequenceAdder,
-    adder_priority=UniformAdderPriority,
-    executor_environment_loop=building.ParallelExecutorEnvironmentLoop,
-    networks=building.DefaultNetworks,
-).get()
-
-
-#########################################################################
-# Test executor in isolation.
-class TestSystemExecutor(System):
-    def design(self) -> Tuple[DesignSpec, Dict]:
-        """Mock system design with zero components.
+from tests.jax.integration.mock_systems import mock_system_single_process
 
-        Returns:
-            system callback components
-        """
-        components = DesignSpec(
-            **system_init,
-            data_server=mocks.MockOnPolicyDataServer,
-            data_server_adder_signature=ParallelSequenceAdderSignature,
-            parameter_server=mocks.MockParameterServer,
-            executor_parameter_client=mocks.MockExecutorParameterClient,
-            trainer_parameter_client=mocks.MockTrainerParameterClient,
-            logger=mocks.MockLogger,
-            **executor,
-            trainer=mocks.MockTrainer,
-            trainer_dataset=mocks.MockTrainerDataset,
-            distributor=mocks.MockDistributor,
-        )
-        return components, {}
+# Environment.
+environment_factory = functools.partial(
+    debugging_utils.make_environment,
+    env_name="simple_spread",
+    action_space="discrete",
+)
 
 
 @pytest.fixture
-def test_exector_system() -> System:
-    """Add description here."""
-    return TestSystemExecutor()
+def test_system_sp() -> System:
+    """A single process built system"""
+    return mock_system_single_process()
 
 
-# Skip failing test for now
-@pytest.mark.skip
-def test_executor(
-    test_exector_system: System,
-) -> None:
-    """Test if the parameter server instantiates processes as expected."""
-
-    # Environment.
-    environment_factory = functools.partial(
-        debugging_utils.make_environment,
-        env_name="simple_spread",
-        action_space="discrete",
-    )
-
-    # Networks.
-    network_factory = ippo.make_default_networks
-
-    # Build the system
-    test_exector_system.build(
-        environment_factory=environment_factory, network_factory=network_factory
-    )
-
+def test_executor_single_process(test_system_sp: System) -> None:
+    """Test if the executor instantiates processes as expected."""
     (
         data_server,
         parameter_server,
         executor,
         evaluator,
         trainer,
-    ) = test_exector_system._builder.store.system_build
+    ) = test_system_sp._builder.store.system_build
 
-    assert isinstance(executor, acme.core.Worker)
+    # _writer.append needs to be called once to get _writer.history
+    # _writer.append called in observe_first and observe
+    with pytest.raises(RuntimeError):
```

Comment:
Why is this with needed?

Reply:
To test that the adder initially don't have any history (was empty at first )

Reply:
So the test passes if the assert fails?

Reply:
yeah because it doesn't even have a ._writer at that point

Reply:
Should I reformulate it?

Reply:
I think its fine. `RuntimeError: history cannot be accessed before `append` is called at least once.` so we are verifying that we have not yet called append

Reply:
Sounds good to me :)